packages/vitest/src/node/pools/rpc.ts

```diff
@@ -20,8 +25,31 @@ export function createMethodsRPC(project: WorkspaceProject): RuntimeRPC {
       const r = await project.vitenode.transformRequest(id)
       return r?.map as RawSourceMap | undefined
     },
-    fetch(id, transformMode) {
-      return project.vitenode.fetchModule(id, transformMode)
+    async fetch(id, transformMode) {
+      const result = await project.vitenode.fetchResult(id, transformMode)
+      const code = result.code
+      if (result.externalize)
+        return result
+      if ('id' in result)
+        return { id: result.id as string }
+
+      if (!code)
+        throw new Error(`Failed to fetch module ${id}`)
+
+      const dir = join(project.tmpDir, transformMode)
+      const tmp = join(dir, id.replace(/[/\\?%*:|"<>]/g, '_').replace('\0', '__x00__'))
```

Comment:
I think this is causing too long filenames:

https://github.com/vitest-dev/vitest-ecosystem-ci/actions/runs/8911833491/job/24473995666

```
⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯
  Error: ENAMETOOLONG: name too long, open '/tmp/a5LDqI7rqUo2IYVisuaPM/ssr/_home_runner_work_vitest-ecosystem-ci_vitest-ecosystem-ci_workspace_elk_elk_node_modules_.pnpm_vitest-environment-nuxt@1.0.0_@vitest+ui@1.5.3_@vue+test-utils@2.4.5_h3@1.11.1_happy-dom@10.5_ftcywtwlwumcv752fah2mtnl3i_node_modules_vitest-environment-nuxt_index.mjs'
```